src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/hdc/HdcApiClient.kt

```diff
@@ -4,19 +4,18 @@ import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.http.MediaType
 import org.springframework.stereotype.Service
 import org.springframework.web.reactive.function.client.WebClient
-import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.HdcLicence
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.typeReference
 
 @Service
 class HdcApiClient(@Qualifier("oauthHdcApiClient") val hdcApiWebClient: WebClient) {
 
-  fun getByBookingId(bookingId: Long): HdcLicence? {
+  fun getByBookingId(bookingId: Long): HdcLicenceData? {
     return hdcApiWebClient
       .get()
       .uri("/licence/hdc/$bookingId")
       .accept(MediaType.APPLICATION_JSON)
       .retrieve()
-      .bodyToMono(typeReference<HdcLicence>())
+      .bodyToMono(typeReference<HdcLicenceData>())
```

Comment:
**suggestion:** worth thinking about the error handling [here](https://github.com/ministryofjustice/hmpps-hdc-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt#L18).

Ideally the hdc api would 404 and this service would propagate the 404 response through to the ui.

Reply:
if the backend api returns a 404 this would throw a 500 response.. might be worth having a look at `coerce404ResponseToNull(it)` and create an equivalent `propagate404Response(it)`